### PR TITLE
BracesFixer - correctly fix indents of anonymous functions/classes

### LINKED ISSUE #1094: Certain Closures Not Being Fixed
I'd expect the following:

```php
<?php

$function = function($foo) { return 'foo'; };
```

to be fixed to:

```php
<?php

$function = function ($foo) {
    return 'foo';
};
```

but instead, it's fixed to:

```php
<?php

$function = function ($foo) { return 'foo'; };
```

ie, there's basically no change (apart from the movement of the bracket just after "function").

### PR DESCRIPTION
**Pull request by gharlan**
closes #1094, refs #1691

### CODE REVIEW

**Comment by gharlan** [comment]
fixed tests and harmonized comment handling around `{`.
Please take a look at changed test cases.

input:
```php
if (true) // foo
{
}

function () // foo
{
}
```

expected before this pr:
```php
if (true) {
    // foo
}

function () {// foo
}
```

harmonized result after merging this pr:
```php
if (true) { // foo
}

function () { // foo
}
```

**Comment by GrahamCampbell** [comment]
I quite liked the comment moving tbh. :P

**Comment by GrahamCampbell** [comment]
Seriously though, your new changes do look much better. ;)

**Comment by gharlan** [comment]
ping :)

**Comment by keradus** [comment]
Anyone willing to make a review ?

File: Symfony/CS/Fixer/PSR2/BracesFixer.php [hunk 1]
```diff
@@ -72,33 +71,43 @@ private function fixCommentBeforeBrace(Tokens $tokens)
...
             }
 
-            $parenthesisEndIndex = $this->findParenthesisEnd($tokens, $index);
-            $afterParenthesisIndex = $tokens->getNextNonWhitespace($parenthesisEndIndex);
-            $afterParenthesisToken = $tokens[$afterParenthesisIndex];
+            $commentIndex = $tokens->getNextNonWhitespace($prevIndex);
+            $commentToken = $tokens[$commentIndex];
 
-            if (!$afterParenthesisToken->isComment()) {
+            if (!$commentToken->isGivenKind(T_COMMENT) || '/*' === substr($commentToken->getContent(), 0, 2)) {
```
Comment (GrahamCampbell) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: how does this cope with random ass missplaced phpdoc?

Comment (gharlan) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: same as with `/*` comments. only `//` and `#` comments need extra handling.

Comment (GrahamCampbell) [inline Symfony/CS/Fixer/PSR2/BracesFixer.php]: Ah, ok. Thanks for explaining. I totally understand now. ;)

**Comment by GrahamCampbell** [comment]
I really wish @DeeJaVu would stop spamming this issue. ;(

**Comment by GrahamCampbell** [comment]
It does look like GitHub have suspended the account though, so that's good. :)

**Comment by keradus** [comment]
Thank you @gharlan.

**Pull request merged**